### PR TITLE
Whack some moles in our test suite

### PR DESCRIPTION
**Pull request by whit537**
- Remove marky-markdown, because we no longer depend on it, and what's more, npm network failures sometimes break the build.
- Make fake packages resilient against name collisions.
- Start debugging an email issue under test cropping up on #4507.

### CODE REVIEW

**Comment by whit537** [comment]
Should be an easy review/merge. cc: @PeterDaveHello @rohitpaulk et al.

**Comment by whit537** [comment]
I cherry-picked f53bf5d3794beea2866e3df14a0374ec44d2d5e7 05a1295195fdff005758c9778aaa9ad7de572dbe and d6496d489c3d7ef63d2320ee0e5826aac1500f90 from #4507 into here.

**Comment by whit537** [comment]
Should still be a pretty easy review/merge.

**Comment by whit537** [comment]
Hah! Got it! I started to see this locally with `self_check` but didn't actually observe `flush`. [Here it is](https://travis-ci.org/gratipay/gratipay.com/builds/242076231#L1075-L1081), though.

```
pid-5223 thread-47101571705216 (MainThread) Cron: installing <lambda> to run every 300 seconds.
pid-5223 thread-47101571705216 (MainThread) Cron: installing self_check to run every 600 seconds with a lock.
pid-5223 thread-47101571705216 (MainThread) Cron: not installing <lambda>.
pid-5223 thread-47101571705216 (MainThread) Cron: installing flush to run every 60 seconds with a lock.
pid-5223 thread-47101571705216 (MainThread) Cron: not installing log_metrics.
pid-5223 thread-47101828482816 (Thread-2) Cron: installing self_check to run every 600 seconds with a lock.
pid-5223 thread-47101828482816 (Thread-2) Cron: installing flush to run every 60 seconds with a lock.
```

**Comment by whit537** [comment]
Alright, so `Application` is getting instantiated twice. That in itself isn't unexpected, I don't think.

**Comment by whit537** [comment]
Looks like we instantiate it in a few places:

- CLIs
- test harness
- wireup
- gunicorn entrypoint

**Comment by whit537** [comment]
So which one are we hitting in `Thread-2`? Where does that thread come from?

File: .travis.yml [hunk 1]
```diff
@@ -30,14 +28,21 @@ cache:
...
     - geckodriver
 install:
  - if [ "${TRAVIS_BRANCH}" = "master" -a "${TRAVIS_PULL_REQUEST}" = "false" ]; then rm -rf env; fi
  - touch requirements.txt package.json
  - make env -j2
 before_script:
  - echo "DATABASE_URL=dbname=gratipay" | tee -a tests/local.env local.env
+
+ # Turn off all cron threads.
+ - echo "UPDATE_CTA_EVERY=0" >> local.env
```
Comment (rohitpaulk) [inline .travis.yml]: Aren't we running with `test.env` overriding `local.env`? If we're already setting these values to `0` in `test.env`, is it required to set them to zero here too?

Comment (whit537) [inline .travis.yml]: All three envvar overrides on this commit are necessary:

- [`make bgrun`](https://github.com/gratipay/gratipay.com/blob/56526b9646938ffa0e4737eb9a04361b0d68f464/.travis.yml#L48) &rarr; [`$(honcho_run)`](https://github.com/gratipay/gratipay.com/blob/56526b9646938ffa0e4737eb9a04361b0d68f464/Makefile#L52) &rarr; [`local.env`](https://github.com/gratipay/gratipay.com/blob/56526b9646938ffa0e4737eb9a04361b0d68f464/Makefile#L12)
- [`make test`](https://github.com/gratipay/gratipay.com/blob/56526b9646938ffa0e4737eb9a04361b0d68f464/.travis.yml#L48) &rarr; [`$(pytest)`](https://github.com/gratipay/gratipay.com/blob/56526b9646938ffa0e4737eb9a04361b0d68f464/Makefile#L67) &rarr; [`$(test_env_files)`](https://github.com/gratipay/gratipay.com/blob/56526b9646938ffa0e4737eb9a04361b0d68f464/Makefile#L13) &rarr; [`tests/test.env`](https://github.com/gratipay/gratipay.com/blob/56526b9646938ffa0e4737eb9a04361b0d68f464/Makefile#L9)
- [`make doc`](https://github.com/gratipay/gratipay.com/blob/56526b9646938ffa0e4737eb9a04361b0d68f464/.travis.yml#L48) &rarr; [`$(doc_env_files)`](https://github.com/gratipay/gratipay.com/blob/56526b9646938ffa0e4737eb9a04361b0d68f464/Makefile#L116) &rarr; [`docs/doc.env`](https://github.com/gratipay/gratipay.com/blob/56526b9646938ffa0e4737eb9a04361b0d68f464/Makefile#L8)

**Pull request merged**